jscomp/runtime/caml_obj.ml

```diff
@@ -223,6 +237,27 @@ and aux_length_b_short (a : Obj.t) (b : Obj.t) i short_length =
     let res = caml_compare (Obj.field a i) (Obj.field b i) in
     if res <> 0 then res
     else aux_length_b_short a b (i+1) short_length
+and aux_obj_compare (a: Obj.t) (b: Obj.t) =
+  let min_key_lhs = ref None in
```

Comment:
Can probably avoid the allocation here if we used Js.Nullable.t?

Reply:
Yes that would work, though this runtime library does not have access to Js.Nullable.
I have a straight js version in https://github.com/BuckleScript/bucklescript/pull/2657/commits/da928dbfbeca7abb912a7f83e1351df0a0bab78b which does that and simplifies a bunch of other things.

Reply:
@bobzhang what do you think?